pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,438 @@
+        """
+        pass_flg = True
+        for key, value in settings['min_array_filter'].items():
+            if dmap_record[key][j] < value:
+                pass_flg = False
+                break
+        for key, value in settings['max_array_filter'].items():
+            if dmap_record[key][j] > value:
+                pass_flg = False
+                break
+
+        for key, value in settings['min_scalar_filter'].items():
+            if dmap_record[key] < value:
+                pass_flg = False
+                break
+
+        for key, value in settings['max_scalar_filter'].items():
+            if dmap_record[key] > value:
+                pass_flg = False
+                break
+
+        for key, value in settings['equal_scalar_filter'].items():
+            if dmap_record[key] != value:
+                pass_flg = False
+                break
+
+        return pass_flg
+
+    @classmethod
+    def __check_data_type(cls, parameter: str, expected_type: str):
+        """
+        Checks to make sure the plot type is correct
+        for the data structure
+
+        Parameters
+        ----------
+        paraneter: str
+            string key word name of the parameter
+        expected_type: str
+            string decsribing an array or scalar type
+            to determine which one is needed for the type of plot
+
+        Raises
+        -------
+        RTPIncorrectPlotMethodError
+        """
+        data_type = cls.dmap_data[0][parameter]
+        if expected_type == 'array':
+            if not isinstance(data_type, np.ndarray):
+                raise rtp_exceptions.RTPIncorrectPlotMethodError(parameter,
+                                                                 data_type)
+        else:
+            if isinstance(data_type, np.ndarray):
+                raise rtp_exceptions.RTPIncorrectPlotMethodError(parameter,
+                                                                 data_type)
+
+    # TODO: move to a utils or superDARN utils
+    @classmethod
+    def __time2datetime(cls, dmap_record: dict) -> datetime:
+        """
+        Converts DMAP time parameter fields into a datetime object
+
+        Parameter
+        ---------
+        dmap_record: dict
+            dictionary of the DMAP data contains the time data
+
+        Returns
+        -------
+        datetime:
+            returns a datetime object of the records time stamp
+        """
+        year = dmap_record['time.yr']
+        month = dmap_record['time.mo']
+        day = dmap_record['time.dy']
+        hour = dmap_record['time.hr']
+        minute = dmap_record['time.mt']
+        second = dmap_record['time.sc']
+        micro_sec = dmap_record['time.us']
+
+        return datetime(year=year, month=month, day=day, hour=hour,
+                        minute=minute, second=second, microsecond=micro_sec)
+
+    # Needs its own method because it generates vertical lines when
+    # the cpid changes
```

Comment:
Are these all `TODO`?

Reply:
HAHA! this is older code. Deleted :p 

However, if we want to have different visuals for CP IDs it would make sense to remove it from time-series. Something to think about!